test/functional/test-amp-img.js

```diff
@@ -373,4 +376,76 @@ describe('amp-img', () => {
     impl.unlayoutCallback();
   });
 
+  describe('blurred image placeholder', () => {
+    beforeEach(() => {
+      toggleExperiment(window, 'blurry-placeholder', true, true);
+    });
+
+    /**
+     * Creates an amp-img with an image child that could potentially be a
+     * blurry placeholder.
+     * @param {boolean} addPlaceholder Whether the child should have a
+     *     placeholder attribute.
+     * @param {boolean} addBlurClass Whether the child should have the
+     *     class that allows it to be a blurred placeholder.
+     * @return {AmpImg} An amp-img object potentially with a blurry placeholder
+     */
+    function getImgWithBlur(addPlaceholder, addBlurClass) {
+      const el = document.createElement('amp-img');
+      const img = document.createElement('img');
+      if (addPlaceholder) {
+        img.setAttribute('placeholder', '');
+        el.getPlaceholder = () => img;
+      } else {
+        el.getPlaceholder = sandbox.stub();
+      }
+      if (addBlurClass) {
+        img.classList.add('i-amphtml-blur');
+      }
+      el.appendChild(img);
+      el.getResources = () => Services.resourcesForDoc(document);
+      const impl = new AmpImg(el);
+      sandbox.stub(impl, 'getLayoutWidth').returns(200);
+      impl.togglePlaceholder = sandbox.stub();
+      return impl;
+    }
+
+    it('should set placeholder opacity to 0 on image load', () => {
+      let impl = getImgWithBlur(true, true);
+      impl.buildCallback();
+      impl.layoutCallback();
+      impl.firstLayoutCompleted();
+      let el = impl.element;
+      let img = el.firstChild;
+      expect(img.style.opacity).to.equal('0');
+      expect(impl.togglePlaceholder).to.not.be.called;
+
+      impl = getImgWithBlur(true, false);
+      impl.buildCallback();
+      impl.layoutCallback();
+      impl.firstLayoutCompleted();
+      el = impl.element;
+      img = el.firstChild;
+      expect(img.style.opacity).to.be.equal('');
+      expect(impl.togglePlaceholder).to.have.been.calledWith(false);
+
+      impl = getImgWithBlur(false, true);
+      impl.buildCallback();
+      impl.layoutCallback();
+      impl.firstLayoutCompleted();
+      el = impl.element;
+      img = el.firstChild;
+      expect(img.style.opacity).to.be.equal('');
+      expect(impl.togglePlaceholder).to.have.been.calledWith(false);
+
+      impl = getImgWithBlur(false, false);
+      impl.buildCallback();
+      impl.layoutCallback();
+      impl.firstLayoutCompleted();
+      el = impl.element;
+      img = el.firstChild;
+      expect(impl.togglePlaceholder).to.have.been.calledWith(false);
+      impl.unlayoutCallback();
```

Comment:
Missed comment from earlier: Why is `unlayoutCallback` called here?

Reply:
Oh sorry! I just saw it in the aftereach of other tests, so I included it in my aftereach method, and then I took out the afterEach because I only had one test method after deleting the other two, so I included the code here. I'll take it out though!